python/cuml/decomposition/pca.pyx

```diff
@@ -387,47 +393,52 @@ class PCA:
-                             <double*> singular_vals_ptr,
-                             <double*> mean_ptr,
-                             <double*> noise_vars_ptr,
-                             params)
+                pcaFit(h_[0],
+                       <double*> input_ptr,
+                       <double*> components_ptr,
+                       <double*> explained_var_ptr,
+                       <double*> explained_var_ratio_ptr,
+                       <double*> singular_vals_ptr,
+                       <double*> mean_ptr,
+                       <double*> noise_vars_ptr,
+                       params)
         else:
 
             if self.gdf_datatype.type == np.float32:
-                pcaFitTransform(<float*> input_ptr,
-                                      <float*> trans_input_ptr,
-                                      <float*> components_ptr,
-                                      <float*> explained_var_ptr,
-                                      <float*> explained_var_ratio_ptr,
-                                      <float*> singular_vals_ptr,
-                                      <float*> mean_ptr,
-                                      <float*> noise_vars_ptr,
-                                      params)
+                pcaFitTransform(h_[0],
+                                <float*> input_ptr,
+                                <float*> trans_input_ptr,
+                                <float*> components_ptr,
+                                <float*> explained_var_ptr,
+                                <float*> explained_var_ratio_ptr,
+                                <float*> singular_vals_ptr,
+                                <float*> mean_ptr,
+                                <float*> noise_vars_ptr,
+                                params)
             else:
-                pcaFitTransform(<double*> input_ptr,
-                                      <double*> trans_input_ptr,
-                                      <double*> components_ptr,
-                                      <double*> explained_var_ptr,
-                                      <double*> explained_var_ratio_ptr,
-                                      <double*> singular_vals_ptr,
-                                      <double*> mean_ptr,
-                                      <double*> noise_vars_ptr,
-                                      params)
+                pcaFitTransform(h_[0],
+                                <double*> input_ptr,
+                                <double*> trans_input_ptr,
+                                <double*> components_ptr,
+                                <double*> explained_var_ptr,
+                                <double*> explained_var_ratio_ptr,
+                                <double*> singular_vals_ptr,
+                                <double*> mean_ptr,
+                                <double*> noise_vars_ptr,
+                                params)
```

Comment:
Line 445 access `self.components_` on the host, right? Does that ensure correct synchronization of the potentially asynchronous `pcaFitTransform` call?

Reply:
`components_` seems to be a device buffer as per line 311. I believe its access on the host is done by a `cudaMemcpy`? @cjnolet ?

Reply:
I think it's better to know first how the copy is done and then decide to put sync here or not, no?

Reply:
I've asked this question to our cudf folks. Awaiting answer...

Reply:
Makes sense. I have issued a sync call.